ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientImpl.java

```diff
@@ -253,8 +270,46 @@ public IngestionResult ingestFromResultSet(ResultSetSourceInfo resultSetSourceIn
         }
     }
 
+    protected void validateEndpointServiceType() throws IngestionServiceException, IngestionClientException {
+        if (StringUtils.isBlank(endpointServiceType) && resourceManager != null) {
+            endpointServiceType = resourceManager.getServiceType();
+        }
+        if (!DATA_MANAGEMENT_SERVICE_TYPE.equals(endpointServiceType)) {
+            String message = String.format(WRONG_ENDPOINT_MESSAGE, DATA_MANAGEMENT_SERVICE_TYPE, endpointServiceType);
+            if (StringUtils.isNotBlank(generateEndpointSuggestion())) {
+                message = String.format(message + ": '%s'", suggestedEndpointUri);
+            } else {
+                message += ".";
+            }
+            throw new IngestionClientException(message);
+        }
+    }
+
+    private String generateEndpointSuggestion() {
+        if (suggestedEndpointUri != null) {
+            return suggestedEndpointUri;
+        }
+        // The default is not passing a suggestion to the exception
+        suggestedEndpointUri = "";
```

Comment:
StringUtils.EMPTY

Reply:
I'll change it if that's our team's convention, but my reasoning behind "" as opposed to StringUtils.EMPTY is explained here: https://stackoverflow.com/questions/3450604/why-is-there-no-string-empty-in-java

Reply:
I just looked it up in our Java SDK codebase, and I'm not sure that StringUtils.EMPTY is our convention there.

StringUtils.EMPTY is used 0 times, and "" is used 67 times.

Let me know if you'd like me to change it anyway.

Reply:
Nvm, keep it this way.